app/timetables/tests/test_models.py

```diff
@@ -65,3 +65,63 @@ def test_meal_end_time_same_with_start_time_cannot_be_saved(self):
         )
 
         self.assertRaises(ValidationError, meal.save)
+
+
+class TimetableTest(TestCase):
+    """Tests the Timetable model."""
+
+    def setUp(self):
+        Timetable.objects.create(
+            name='timtable-item',
+            code='FBI23212',
+            api_key='419223',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('5:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('10:7:9', '%H:%M:%S'))
+        )
+    
+    def test_timetable_name_should_be_capitalized_on_save(self):
+        timetable = Timetable.objects.get(name__iexact='timtable-item')
+        
+        self.assertEqual(timetable.name, 'Timtable-item')
+
+    def test_duplicate_timetable_name_cannot_be_saved(self):
+        timetable = Timetable(
+            name='timtable-item',
+            code='FB23212',
+            api_key='41923',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('4:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('15:7:9', '%H:%M:%S'))
+        )
+
+        self.assertRaises(IntegrityError, timetable.save)
+
+    def test_duplicate_timetable_code_cannot_be_saved(self):
+        timetable = Timetable(
+            name='timtable-value',
+            code='FBI23212',
+            api_key='41923',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('4:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('15:7:9', '%H:%M:%S'))
+        )
+        
+        self.assertRaises(IntegrityError, timetable.save)
+    
+    def test_duplicate_timetable_code_cannot_be_saved(self):
+        timetable = Timetable(
+            name='timtable',
+            code='FBI232123',
+            api_key='419223',
+            cycle_length=1,
+            current_cycle_day=2,
+            date_created=datetime.time(datetime.strptime('4:7:9', '%H:%M:%S')),
+            date_modified=datetime.time(datetime.strptime('15:7:9', '%H:%M:%S'))
+        )
+        
+        self.assertRaises(IntegrityError, timetable.save)
+
```

Comment:
@andela-joyebanji please add a test case to ensure that if a cycle_length is 1, you cannot have a current_cycle_day of a number higher than the cycle_length like 2. 
```
Timetables in Taverna are best suited for cafeteria systems where the menu is repeated after a period of time. This period, measured in days, is known as the cycle_length of the timetable. For better context, Andela Lagos’ timetable has a cycle length of 14 days (2 weeks) after which the food schedule is repeated. In this case,
Monday of the first week is Cycle Day 1, Friday of the first week is Cycle Day 5, Sunday of the first week is Cycle Day 7.
Monday of the second week is Cycle Day 8, Friday of the second week is Cycle Day 12, Sunday of the second week is Cycle Day 14. Monday of the third week is Cycle Day 1 (again). 
```
So if our Andela's cycle length is currently 14, we cannot have a current_cycle_day of 15.